lib/xdrgen/generators/rust/src/types.rs

```diff
@@ -1219,8 +1222,15 @@ impl<const MAX: u32> ReadXdr for VecM<u8, MAX> {
                 return Err(Error::LengthExceedsMax);
             }
 
-            let mut vec = vec![0u8; len as usize];
-            r.read_exact(&mut vec)?;
+            let mut vec = vec![0u8; 0];
+            let mut len_remaining = len as usize;
+            while len_remaining > 0 {
```

Comment:
It seems like it should be easy to factor this out to a helper function. It's way too easy to mess up when updating the same code in three different places.